File: src/GS/cmds/ListCmd.cxx
/*                     L I S T C M D . C X X
 * BRL-CAD
 *
 * Copyright (c) 2011 United States Government as represented by
 * the U.S. Army Research Laboratory.
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public License
 * version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this file; see the file named COPYING for more
 * information.
 */
/** @file ListCmd.cxx
 *
 * Brief description
 *
 */

#include "ListCmd.h"
#include "Portal.h"
#include "PortalManager.h"
#include "TypeOnlyMsg.h"
#include "NetMsgTypes.h"
#include "DirListReqMsg.h"

ListCmd::ListCmd() : AbstractClientCmd("list") {}

ListCmd::~ListCmd() {}

std::string
ListCmd::getUsage()
{
    return "Usage: list pathToGeometry";
}

std::string
ListCmd::getHelp()
{
    return "Attempts to get a list of geometry for the provided path from the current GeometryService";
}

bool
ListCmd::_exec(GSCmdLineClient* client, std::list<std::string> args)
{
    int argn = args.size();

    if (argn != 1) {
    	this->printUsage();
    	return false;
    }

    /* Check to see if we are connected */
    Portal* p = client->getCurrentPortal();
    if (p == NULL) {
    	this->log->logWARNING("ListCmd", "Not connected to a Geometry Service.");
		return false;
    }
    std::list<std::string>::iterator it = args.begin();
    std::string path(*it);

    this->log->logINFO("ListCmd", "Attempting to list: '" + path + "'.");

    DirListReqMsg req(path);

    p->send(&req);

    return true;
}

/*
 * Local Variables:
 * tab-width: 8
 * mode: C
 * indent-tabs-mode: t
 * c-file-style: "stroustrup"
 * End:
 * ex: shiftwidth=4 tabstop=8
 */

File: src/clients/java/src/org/brlcad/geometryservice/minimalclient/cmd/ListCmd.java
/*
 * BRL-CAD
 *
 * Copyright (c) 2011 United States Government as represented by
 * the U.S. Army Research Laboratory.
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public License
 * version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this file; see the file named COPYING for more
 * information.
 */
/** @file HelpCmd.java
 *
 */
package org.brlcad.geometryservice.minimalclient.cmd;

import java.util.ArrayList;

import org.brlcad.geometryservice.GSJavaInterface;
import org.brlcad.geometryservice.GSStatics;
import org.brlcad.geometryservice.minimalclient.CmdConsolePanel;

/**
 * @author david.h.loman
 *
 */
public class ListCmd extends AbstractCmd {

	/**
	 * @param cmd
	 * @param cmdConsole
	 */
	public ListCmd(CmdConsolePanel cmdConsole) {
		super("list", cmdConsole);
	}

	/*
	 * (non-Javadoc)
	 *
	 * @see
	 * org.brlcad.geometryservice.minimalclient.cmd.AbstractCmd#doCmd(java.lang
	 * .String[])
	 */
	@Override
	public boolean doCmd(String[] args, GSJavaInterface gsji) {

		if (args.length <= 2) {
			this.printUsage();
			return false;
		}

		/* validate Args */
		String path = "/";

		/* if a path is provided, use it.  If not, assume root or "/" */
		if (args.length == 2) {
			path = args[1];
			if (path.length() < 1) {
				GSStatics.stdErr.println("Path too short.");
				this.printUsage();
				return false;
			}
		}
		
		ArrayList<String> items = gsji.getList(path);
		
		this.cmdConsole.printLnToConsole(GSStatics.tab2x + "Got: " + items.size() + " items.", CmdConsolePanel.STYLE_BLUE_BOLD);
		
		return true;
	}

	/*
	 * (non-Javadoc)
	 *
	 * @see org.brlcad.geometryservice.minimalclient.cmd.AbstractCmd#getHelp()
	 */
	@Override
	public String getHelp() {
		return "Lists the contents of a directory";
	}

	/*
	 * (non-Javadoc)
	 *
	 * @see org.brlcad.geometryservice.minimalclient.cmd.AbstractCmd#getUsage()
	 */
	@Override
	public String getUsage() {
		return "list path";
	}

}
